ui/components/GithubDeviceAuthModal.tsx

```diff
@@ -26,11 +26,23 @@ type Props = {
 const Pad = styled(Flex)`
   padding: 8px 0;
 `;
-
+const PointerIcon = styled(Icon)`
+  cursor: pointer;
+`;
 const ModalContent = styled(({ codeRes, onSuccess, onError, className }) => {
   // Move this to a component so that we get the cancel logic when the modal closes.
   const { getGithubAuthStatus } = useAuth();
   const [loading, setLoading] = React.useState(true);
+  const [copied, setCopied] = React.useState(false);
+
+  const handleCopy = React.useCallback(() => {
+    navigator.clipboard.writeText(codeRes.userCode).then(() => {
+      setCopied(true);
+      setTimeout(() => {
```

Comment:
❔ Would it better for this to have a `.then` and a `.catch` instead of the `setTimeout`? We could use `notifyError` that way. This definitely returns a promise and I didn't treat it that way in `YamlView` - you'd be doing me a huge favor if you could move this function to `.utils` and export it to both this component and YamlView!!

Reply:
`setTimeout` here is meant to change the icon back to copy instead of notifying the user with a copied message